functions/helloworld/hello-pub-sub/src/test/java/functions/ExampleIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+package functions;
+
+// [START functions_pubsub_integration_test]
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.github.resilience4j.core.IntervalFunction;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import io.github.resilience4j.retry.RetryRegistry;
+import io.vavr.CheckedRunnable;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.UUID;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ExampleIntegrationTest {
+  // Root URL pointing to the locally hosted function
+  // The Functions Framework Maven plugin lets us run a function locally
+  private static final String BASE_URL = "http://localhost:8080";
+
+  private static Process emulatorProcess = null;
+  private static HttpClient client = HttpClientBuilder.create().build();
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    // Emulate the function locally by running the Functions Framework Maven plugin
+    emulatorProcess = (new ProcessBuilder()).command("mvn", "function:run").start();
```

Comment:
We could likely move the retry logic here instead, and wait until the process has "warmed up".

That said, the HTTP PR (#2691) was already merged using this approach. IMO we should merge as-is (to avoid blocking TW) and push a fast-follow fix.

Reply:
That PR was _just_ merged 🙃. I'll let this slide provided you open an issue on java-docs-samples and assign it to yourself. FWIW pushing a bad sample to avoid blocking a TW is not a good reason. 

I don't think retry logic is needed at all. You can start up the process and wait for it to print "ready to serve" or whatever signal it gives that it's ready to start up.

We should save retries for legitimately flakey services. If starting the emulator startup is flakey, that's a different problem.

Reply:
See #2763 - would you mind chiming in and making a suggestion on which solution to use there?